system/fs_mock.go

```diff
@@ -149,6 +149,7 @@ func (fs *MockFilesystem) stat(name string) (os.FileInfo, error) {
 			log.Println("A")
 			return nil, err
 		}
+		defer tmpFile.Close()
```

Comment:
```suggestion
		defer func() {
			tmpFile.Close()
			os.Remove(tmpFile.Name())
		}()
```
and remove line 153 (I'm not able to mark both lines for the suggestion)